src/parser/statement.js

```diff
@@ -303,31 +304,39 @@ pp.parseForStatement = function (node) {
     return this.parseFor(node, init);
   }
 
+  let usedNowKeyword = false;
+  if (this.hasPlugin("lightscript") && (!forAwait)) {
+    // idx, elem, key, or val begins a LS-enhanced loop. The lookahead token must be a
+    // var name. If it isn't, someone is using one of the contextual keywords as a var name.
+    if (
+      (this.isContextual("idx") || this.isContextual("elem") || this.isContextual("key") || this.isContextual("val")) &&
+      (this.lookahead().type === tt.name)
```

Comment:
@rattrayalex 

This change was bad.
```
for (key of iterable) {}
```
doesn't compile now.

I added this as a test case. I think we still need to lookahead for `of` here, unless you can think of something else.